quickwit-cluster/src/cluster.rs

```diff
@@ -363,28 +363,22 @@ mod tests {
     async fn test_cluster_multiple_node() -> anyhow::Result<()> {
         let tmp_dir = tempfile::tempdir()?;
 
-        let host_key_path1 = tmp_dir.path().join("host_key1");
-        let host_key1 = read_host_key(host_key_path1.as_path())?;
-        let tmp_port1 = available_port()?;
-        let addr_str1 = format!("127.0.0.1:{}", tmp_port1);
-        let remote_host1: SocketAddr = addr_str1.as_str().parse()?;
-        let cluster1 = Cluster::new(host_key1, remote_host1)?;
-
-        let host_key_path2 = tmp_dir.path().join("host_key2");
-        let host_key2 = read_host_key(host_key_path2.as_path())?;
-        let tmp_port2 = available_port()?;
-        let addr_str2 = format!("127.0.0.1:{}", tmp_port2);
-        let remote_host2: SocketAddr = addr_str2.as_str().parse()?;
-        let cluster2 = Cluster::new(host_key2, remote_host2)?;
-        cluster2.add_peer_node(remote_host1);
-
-        let host_key_path3 = tmp_dir.path().join("host_key3");
-        let host_key3 = read_host_key(host_key_path3.as_path())?;
-        let tmp_port3 = available_port()?;
-        let addr_str3 = format!("127.0.0.1:{}", tmp_port3);
-        let remote_host3: SocketAddr = addr_str3.as_str().parse()?;
-        let cluster3 = Cluster::new(host_key3, remote_host3)?;
-        cluster3.add_peer_node(remote_host1);
+        let host_key1 = read_host_key(tmp_dir.path().join("host_key1").as_path())?;
+        let listen_addr1 =
+            SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), available_port()?);
+        let cluster1 = Arc::new(Cluster::new(host_key1, listen_addr1)?);
```

Comment:
I still think this should be a utility function returning a test cluster, the listen address can be accesses via the the cluster object

Reply:
Thanks, I added a test_utils module.
Please take another look.
https://github.com/quickwit-inc/quickwit/pull/323/commits/5c2953f024e7e81b81b83e6409133fcc67f926f4